indexing-lambda/src/main/java/edu/nix/IndexingLambda.java

```diff
@@ -0,0 +1,95 @@
+import java.util.Map;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.apache.solr.common.SolrInputDocument;
+
+public class IndexingLambda {
+    private static final String SOLR_EC2_URL =
+        "http://ec2-13-50-246-93.eu-north-1.compute.amazonaws.com:8983/solr/books";
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final BasicAWSCredentials BASIC_AWS_CREDENTIALS =
+        new BasicAWSCredentials("accessKey", "secretKey");
+
+    public String handleRequest(S3Event input) {
+        String bucketName = input.getRecords().get(0).getS3().getBucket().getName();
+        String key = input.getRecords().get(0).getS3().getObject().getKey();
+
+        AmazonS3 amazonS3 = AmazonS3ClientBuilder.standard()
+            .withCredentials(new AWSStaticCredentialsProvider(BASIC_AWS_CREDENTIALS))
+            .withRegion("eu-north-1").build();
+
+        String document;
+        try {
+            document = getS3Object(amazonS3.getObject(new GetObjectRequest(bucketName, key)));
+        } catch (IOException e) {
+            throw new RuntimeException(e + "Bucket " + bucketName + " wasn't found!");
+        }
+        indexDocument(document);
+
+        return "Document has been indexed successfully!";
+    }
+
+    private void indexDocument(String document) {
+        List<SolrInputDocument> solrInputDocument = jsonDocToSolrDoc(document);
+        SolrClient solrClient = new HttpSolrClient.Builder(SOLR_EC2_URL).build();
+        try {
+            solrClient.add(solrInputDocument);
+            solrClient.commit();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String getS3Object(S3Object s3Object) throws IOException {
+        BufferedReader reader = new BufferedReader(
+            new InputStreamReader(s3Object.getObjectContent().getDelegateStream()));
+        StringBuilder jsonDocument = new StringBuilder();
+        String line = reader.readLine();
+
+        while (line != null) {
+            jsonDocument.append(line);
+            line = reader.readLine();
+        }
+        return jsonDocument.toString();
+    }
+
+    private List<SolrInputDocument> jsonDocToSolrDoc(String jsonDoc) {
+        List<Map<String, String>> documents;
+        List<SolrInputDocument> solrInputDocuments = new ArrayList<>();
+        TypeReference<List<Map<String, String>>> typeReference = new TypeReference<>() {};
+
+        try {
+            documents = OBJECT_MAPPER.readValue(jsonDoc, typeReference);
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException(e);
+        }
+        for (Map<String, String> document : documents) {
+            SolrInputDocument solrInputDocument = new SolrInputDocument();
+            document.keySet().forEach(field ->
+                addFieldsSolrInputDocument(solrInputDocument, field, document.get(field)));
```

Comment:
I don't think we need a separate method for this. You can use:
`field -> solrInputDocument.addField(field, inputDocument.get(field)`